boat_simulator/nodes/physics_engine/fluid_forces.py

```diff
@@ -0,0 +1,93 @@
+"""This module provides functionality for computing the lift and drag forces acting on a medium."""
+
+from typing import Tuple
+
+from numpy.typing import NDArray
+
+from boat_simulator.common.types import Scalar
+
+
+class MediumForceComputation:
+    """This class calculates the lift and drag forces experienced by a medium when subjected to
+    fluid flow.
+
+    Attributes:
+        `lift_coefficients` (NDArray): An array of shape (n, 2) where each row contains a pair
+            (x, y) representing an angle of attack, in degrees, and its corresponding lift
+            coefficient.
+        `drag_coefficients` (NDArray): An array of shape (n, 2) where each row contains a pair
+            (x, y) representing an angle of attack, in degrees, and its corresponding drag
+            coefficient.
+        `areas` (NDArray): An array of shape (n, 2) where each row contains a pair (x, y),
+            representing an angle of attack, in degrees, and its corresponding area, in square
+            meters (m^2).
+        `fluid_density` (Scalar): The density of the fluid acting on the medium, expressed in
+            kilograms per cubic meter (kg/m^3).
+    """
+
+    def __init__(
+        self,
+        lift_coefficients: NDArray,
+        drag_coefficients: NDArray,
+        areas: NDArray,
+        fluid_density: Scalar,
+    ):
+        self.__lift_coefficients = lift_coefficients
+        self.__drag_coefficients = drag_coefficients
+        self.__areas = areas
+        self.__fluid_density = fluid_density
+
+    def compute(self, apparent_velocity: NDArray, orientation: Scalar) -> Tuple[NDArray, NDArray]:
+        """Computes the lift and drag forces experienced by a medium immersed in a fluid.
+
+        Args:
+            apparent_velocity (NDArray): The apparent (relative) velocity between the fluid and the
+                medium, calculated as the difference between the fluid velocity and the medium
+                velocity (fluid_velocity - medium_velocity), expressed in meters per second (m/s).
+            orientation (Scalar): The orientation angle of the medium, expressed in degrees,
+                indicating its alignment in a particular reference frame.
```

Comment:
Not sure what you mean by "indicating its alignment in a particular reference frame".

One thing to note is that in general, when we talk about Euler angles, they can be somewhat arbitrary because Euler angles can follow different conventions (i.e. where is 0 degrees? Do angles increase CW or CCW?). Therefore, I suggest whenever you talk about Euler angles, try to specify the angle convention (if it makes sense to) used to avoid confusion.

The convention we follow is 0 degrees on the positive x-axis and angles increase CCW.